mongodb/db.go

```diff
@@ -162,39 +171,56 @@ func (db *MongoDB) Iterator(start, end []byte) (tmdb.Iterator, error) {
 	if (start != nil && len(start) == 0) || (end != nil && len(end) == 0) {
 		return nil, tmdb.ErrKeyEmpty
 	}
-	filter := bson.M{"key": bson.M{
-		"$gte": start,
-		"$lt":  end,
-	}}
+	cond := bson.M{}
+	if start != nil {
+		cond["$gte"] = makekey(start)
+	}
+	if end != nil {
+		cond["$lt"] = makekey(end)
+	}
+	filter := bson.M{"key": cond}
 
 	findOptions := options.Find()
-	findOptions.SetSort(bson.D{{"key", 1}})
+	findOptions.SetSort(bson.M{"key": 1})
+	fmt.Printf("Mongo Iterator %s %s", hex.EncodeToString(start), hex.EncodeToString(end))
 
 	cursor, err := db.collection.Find(context.Background(), filter, findOptions)
 	if err != nil {
 		return nil, err
 	}
 
-	return newMongoDBIterator(cursor, start, end), nil
+	itr := newMongoDBIterator(cursor, start, end)
+	if itr.Key() == nil {
+		itr.Next()
+	}
```

Comment:
Can you help me understand why we need to handle this case?

Reply:
it seems the iterator in tm-db is slightly different with mongo cursor. The iterator must point to the first result in tm-db, while  mongo cursor is point to nil after it was created, so we need call next to point it to the first result.

Reply:
Got it. Thanks.